lib/board.rb

```diff
@@ -4,49 +4,52 @@ class Board
   PLAYER_ONE = 1
   PLAYER_TWO = 2
 
-  attr_reader :errors, :positions
+  attr_reader :error, :positions
 
   def initialize(positions = EMPTY_BOARD)
-    @errors = []
+    @error = nil
     @positions = positions
   end
 
   def available_positions
-    available = []
-    @positions.each_with_index { |pos, idx| available << -pos * (idx + 1) }
-    available.reject { |pos| pos < 0 }
+    @positions.map.with_index { |position, index| index if position < 0 }.reject(&:nil?)
```

Comment:
it's not immediately obvious to me that being greater than zero results in a non-empty position. it feels like an implementation detail. maybe an `is_empty?` method or something?

Reply:
also, it's a bit awkward to `map` with an `if` statement (which returns nil if the condition is false) and then `reject` the nil values. this might be better expressed with `select`

"get all the positions that aren't empty" is a heck of a lot easier to follow than "map over all the positions, grab their index if their value is less than 0, and then reject the indices that weren't grabbed"

Reply:
Yeah, I tried with a select but couldn't get the index value out. I've refactored the board to use a different scheme for available positions, so this method looks a lot cleaner now. Thanks 👍